### PR TITLE
http-lib: add backtrace to logs on connection without response

### PR DESCRIPTION
**Pull request by psafont**
Some long-running migrations stop because of a loss of connection, log more information when it happens.

I couldn't find a way to get the backtrace to be printed in a nice way without adding too much code, this also makes the change backportable.

I would prefer to log it at a debug level, but the function doesn't expose it, and it would complicate backpoerting as well.

### CODE REVIEW

**Review by robhoes — APPROVED** [review]

**Comment by psafont** [comment]
Backport at https://github.com/xapi-project/xen-api-libs-transitional/pull/117

**Review by lindig — APPROVED** [review]

**Review by snwoods — APPROVED** [review]

**Comment by psafont** [comment]
I'm changing the tests around the http client and server to try to test this change

**Comment by psafont** [comment]
Note that the D.Debug is not printed in the default configuration, but the backtraces will be printed. I believe this should not not usually cause logspam as connections are only established to known servers, so parse error (and other unknown exceptions) should be rare in practice/

File: ocaml/libs/http-lib/dune [hunk 1]
```diff
@@ -133,14 +133,10 @@
...
 )
 
-(rule
-  (alias runtest)
+(cram
   (package xapi)
   (deps
     test_client.exe
     test_server.exe
-    client_server_test.sh
```
Comment (robhoes) [inline ocaml/libs/http-lib/dune:142]: how does it find the new test_client_server.t?

Comment (psafont) [reply ocaml/libs/http-lib/dune:142]: The `cram` atom processes all the `.t` files in the directory

**Review by robhoes — APPROVED** [review]

**Review by lindig — APPROVED** [review]

**Pull request merged**